Design.md

```diff
@@ -303,22 +303,60 @@ Get S3 capabilities of the platform
-1. We don’t know yet what needs to be done here
-1. For next generation,
-    1. Utapi Service level -> global platform
-    1. Utapi Account level -> tenant level
-    1. Utapi User level -> user level
 
+### getOsisUsage
+Get the platform usage of global level (without query parameter), tenant level (only with tenant_id) or user  level (with tenant_id and user_id).
+
+Platform usage has the 5 metrics: bucket count, object count, total_bytes, available_bytes, used_bytes at the level of global, tenant and user respectively.
+
+- bucket_count 
+  - global: get all accounts from VAULT which are associated with VCD tenants, 
+  and call OSIS getBucketList API of each tenant respectively to get the bucket count 
+  - tenant: call OSIS getBucketList API of the provided tenant_id to get the bucket count 
+  - user: call OSIS getBucketList API of the provided tenant_id to get all buckets then filter by bucket owner equal to provided user_id
+
+- object_count 
+  - global: get all accounts from VAULT which are associated with VCD tenants, 
+  call UTAPI ListMetrics API with all tenant_ids to get `numberOfObjects` field. 
+  (Split into multiple requests if the number of accounts is big, the threshold value is to be defined) 
+  - tenant: call UTAPI ListMetrics API with tenant_id to get `numberOfObjects` field 
+  - user: call UTAPI ListMetrics API with user_id to get `numberOfObjects` field
+
+- total_bytes 
+  - tenant: call VAULT getAccount API to get quota of the tenant, total_bytes will be the quota of the current tenant if quota exists, otherwise, it will be 0 
+  - global and user: 0
+
+- used_bytes 
+  - global: call UTAPI ListMetrics with all tenant_ids to get `storageUtilized` filed. (Split into multiple requests if the number of accounts is big, the threshold value is to be defined)
+  - tenant: call UTAPI ListMetrics with tenant_id to get `storageUtilized` field 
+  - user: call UTAPI ListMetrics with user_id to get `storageUtilized` field
+
+- available_bytes 
+  - tenant: can be calculated by total_bytes and used_bytes if total_bytes exists, otherwise, will be 0 as well. 
+  - global and user: 0
+
+#### add UTAPI endpoint and UTAPI client
+Like how we integrated Vault/Cloudserver with OSIS, add UTAPI host and port in `application.properties` file, create a UtapiClient class and add a listMetrics function to call Utapi listMetrics requests.
+
+
+#### with Utapi enabled or disabled
+To not increase dependency complexity,  we will not make UTAPI mandatory for running OSIS as this getUsage API is optional for OSE working with OSIS, if UTAPI is not enabled alongside OSIS, we won’t return any error but 0 metrics.
+
+We will have a config field `osis.scality.utapi.enabled` in `application.properties` which can automatically read from [Federation group_vars file](https://github.com/scality/Federation/blob/development/7.10/env/light-local/group_vars/all#L146) about if UTAPI is enable or not in S3C cluster, we can as well configure it manually.
```

Comment:
The details of complexity of federation is not required here. (unless given as an example)
Same, OSIS should work with Federation and ARTESCA in the future.
Hence in Artesca's case an operator will detect it.

If UTAPI is enabled and not available, then what do we do?
In case of failure of UTAPI service, do we return an error?

Reply:
will remove this part.
if UTAPI is enabled, then we communicate with UTAPI to get its metrics, if UTAPI is not enabled or not reachable or returns any failure, OSIS won't return error but 0 metrics. because according to [vmware osis doc](https://developer.vmware.com/apis/1034#/usage/getUsage), the response code for getUsage API is only 200 and 501(not implemented)

Reply:
fixed: https://github.com/scality/osis/pull/115/commits/b5dea5a9331804bc5482a7ece9d4e57eb8c1183a